pyingest/parsers/arxiv.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+
+# arxiv_dc: parser for arXiv records in Dublin Core XML format
+# M. Templeton, 2017 November 16
+
+import codecs
+from dubcore import DublinCoreParser
+from adsputils import u2asc
+from xml.parsers import expat
+
+
+class MissingAuthorException(Exception):
+    pass
+
+class MissingTitleException(Exception):
+    pass
+
+class MissingAbstractException(Exception):
+    pass
+
+class MissingIDException(Exception):
+    pass
+
+class MissingDateException(Exception):
+    pass
+
+class EmptyParserException(Exception):
+    pass
+
+class ArxivParser(DublinCoreParser):
+
+    def get_author_init(self,namestring):
+        output = u2asc(namestring)
+        for c in output:
+            if c.isalpha():
+                return c
+        return u'.'
+
+
+    def parse(self, fp, **kwargs):
+
+        arx = dict()
+
+        try:
+            r = super(self.__class__, self).parse(fp, **kwargs)
+        except expat.ExpatError:
+            raise EmptyParserException("Not parseable xml")
+        else:
+            pass
+
+        if r['abstract']:
+            r['abstract']=r['abstract'][0]
+
+        if r['bibcode']:
+            idarray = r['bibcode'].split(':')
+            arxiv_id = idarray[-1]
+
+            if arxiv_id[0].isalpha():
+                arx_field = arxiv_id.split('/')[0].replace('-','.')
+                arx_num = arxiv_id.split('/')[1]
+                arx_yy = arx_num[0:2]
+                if arx_num[2] == '0':
+                    arx_num = arx_num[3:]
+                else:
+                    arx_num = arx_num[2:]
+            else:
+                arx_field = 'arXiv'
+                arx_num = arxiv_id.replace('.','')
```

Comment:
I think this does not do the right thing for the "4.4" format of arXiv records, eg. arXiv:0704.0010, please double check